scirpy/datasets/__init__.py

```diff
@@ -205,3 +205,106 @@ def vdjdb(cached: bool = True, *, cache_path="data/vdjdb.h5ad") -> AnnData:
+    ----------
+    cached
+        If `True`, attempt to read from the `data` directory before downloading
+
+    Returns
+    -------
+    An anndata object containing all entries from IEDB in `obs`.
+    Each entry is represented as if it was a cell, but without gene expression.
+    Metadata is stored in `adata.uns["DB"]`.
+    """
+    if cached:
+        try:
+            return sc.read_h5ad(cache_path)
+        except OSError:
+            pass
+
+    tcr_table = pd.read_csv(
+        "https://www.iedb.org/downloader.php?file_name=doc/receptor_full_v3.zip",
+        index_col=None,
+        sep=",",
+        na_values=["None"],
+        true_values=["True"],
+    )
+
+    tcr_table.loc[
+        tcr_table["Chain 1 CDR3 Curated"].isna(), "Chain 1 CDR3 Curated"
+    ] = tcr_table["Chain 1 CDR3 Calculated"]
+
+    tcr_table.loc[
+        tcr_table["Chain 2 CDR3 Curated"].isna(), "Chain 2 CDR3 Curated"
+    ] = tcr_table["Chain 2 CDR3 Calculated"]
+
+    tcr_table_T = tcr_table[(tcr_table["Response Type"] == "T cell")]
+
+    tcr_table_T["Chain 1 CDR3 Curated"] = tcr_table_T[
+        "Chain 1 CDR3 Curated"
+    ].str.upper()
+    tcr_table_T["Chain 2 CDR3 Curated"] = tcr_table_T[
+        "Chain 2 CDR3 Curated"
+    ].str.upper()
+
+    tcr_table = tcr_table_T
+
+    tcr_cells = []
+    for _, row in tcr_table.iterrows():
+        cell = AirrCell(cell_id=row["Receptor ID"])
+        alpha_chain = AirrCell.empty_chain_dict()
+        beta_chain = AirrCell.empty_chain_dict()
+        alpha_chain.update(
+            {
+                "locus": "TRA",
+                "junction_aa": row["Chain 1 CDR3 Curated"],
+                "junction": None,
+                "consensus_count": None,
+                "v_call": row["Curated Chain 1 V Gene"],
+                "j_call": row["Curated Chain 1 J Gene"],
+                "productive": True,
+            }
+        )
+        beta_chain.update(
+            {
+                "locus": "TRB",
+                "junction_aa": row["Chain 2 CDR3 Curated"],
+                "junction": None,
+                "consensus_count": None,
+                "v_call": row["Curated Chain 2 V Gene"],
+                "d_call": row["Curated Chain 2 D Gene"],
+                "j_call": row["Curated Chain 2 J Gene"],
+                "productive": True,
+            }
+        )
+        cell.add_chain(alpha_chain)
+        cell.add_chain(beta_chain)
+        tcr_cells.append(cell)
+
+    logging.info("Converting to AnnData object")
+    iedb = from_airr_cells(tcr_cells)
+    tcr_table = tcr_table.set_index(iedb.obs.index)
+    iedb.obs["Antigen"] = tcr_table["Antigen"]
+    iedb.obs["Organism"] = tcr_table["Organism"]
+    iedb.obs["Chain 1 Type"] = tcr_table["Chain 1 Type"]
+    iedb.obs["Chain 2 Type"] = tcr_table["Chain 2 Type"]
+    iedb.obs["Response Type"] = tcr_table["Response Type"]
+    iedb.obs["Reference IRI"] = tcr_table["Reference IRI"]
+    iedb.obs["Epitope IRI"] = tcr_table["Epitope IRI"]
```

Comment:
you could assign these directly in the previous `for` loop to the `cell` object: 
```python
cell = AirrCell(cell_id=row["Receptor ID"])
cell["Antigen"] = row["Antigen"]
```

that's a bit easier to follow and eliminates an additional error source.